docker/dependencies/Dockerfile

```diff
@@ -0,0 +1,298 @@
+    ldconfig; \
+    rm -rf /tmp/cmake
+
+# install boost 1.65.1
+RUN set -e; \
+    git clone https://github.com/boostorg/boost /tmp/boost; \
+    (cd /tmp/boost ; git checkout 436ad1dfcfc7e0246141beddd11c8a4e9c10b146); \
+    (cd /tmp/boost ; git submodule update --init --recursive); \
+    (cd /tmp/boost ; /tmp/boost/bootstrap.sh --with-libraries=system,filesystem); \
+    (cd /tmp/boost ; /tmp/boost/b2 headers); \
+    (cd /tmp/boost ; /tmp/boost/b2 cxxflags="-std=c++14" -j ${PARALLELISM} install --prefix=/opt/dependencies/boost); \
+    ldconfig; \
+    rm -rf /tmp/boost
+
+# install protobuf v3.5.1
+RUN set -e; \
+    git clone https://github.com/google/protobuf /tmp/protobuf; \
+    (cd /tmp/protobuf ; git checkout 106ffc04be1abf3ff3399f54ccf149815b287dd9); \
+    cmake \
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+        -Dprotobuf_BUILD_TESTS=OFF \
+        -Dprotobuf_BUILD_SHARED_LIBS=ON \
+        -H/tmp/protobuf/cmake \
+        -B/tmp/protobuf/.build \
+        -DCMAKE_INSTALL_PREFIX=/opt/dependencies/protobuf; \
+    cmake --build /tmp/protobuf/.build --target install -- -j${PARALLELISM}; \
+    export LD_LIBRARY_PATH=/opt/dependencies/protobuf/lib:$LD_LIBRARY_PATH; \
+    ldconfig; \
+    rm -rf /tmp/protobuf
+
+# install gflags
+RUN set -e; \
+    git clone https://github.com/gflags/gflags /tmp/gflags; \
+    (cd /tmp/gflags ; git checkout f8a0efe03aa69b3336d8e228b37d4ccb17324b88); \
+    cmake \
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+        -H/tmp/gflags \
+        -B/tmp/gflags/build \
+        -DCMAKE_INSTALL_PREFIX=/opt/dependencies/gflags; \
+    cmake --build /tmp/gflags/build --target install -- -j${PARALLELISM}; \
+    ldconfig; \
+    rm -rf /tmp/gflags
+
+# install c-ares
+RUN set -e; \
+    git clone https://github.com/c-ares/c-ares /tmp/c-ares; \
+    (cd /tmp/c-ares ; git checkout 3be1924221e1326df520f8498d704a5c4c8d0cce); \
+    cmake \
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+        -H/tmp/c-ares \
+        -B/tmp/c-ares/build \
+        -DCMAKE_INSTALL_PREFIX=/opt/dependencies/c-ares; \
+    cmake --build /tmp/c-ares/build --target install -- -j${PARALLELISM}; \
+    ldconfig; \
+    rm -rf /tmp/c-ares
+
+# install grpc 1.11.0
+RUN set -e; \
+    git clone https://github.com/grpc/grpc /tmp/grpc; \
+    (cd /tmp/grpc ; git checkout bd44e485f69d70ca4095cea92decd98de3892aa6); \
+    export LD_LIBRARY_PATH="/opt/dependencies/protobuf/lib:$LD_LIBRARY_PATH"; \
+    cmake \
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+        -DgRPC_BENCHMARK_PROVIDER="" \
+        -DgRPC_ZLIB_PROVIDER=package \
+        -DgRPC_CARES_PROVIDER=package \
+        -DgRPC_SSL_PROVIDER=package \
+        -DgRPC_PROTOBUF_PROVIDER=package \
+        -DgRPC_GFLAGS_PROVIDER=package \
+        -DBUILD_SHARED_LIBS=ON \
+        -H/tmp/grpc \
+        -B/tmp/grpc/.build \
+        -DCMAKE_PREFIX_PATH="/opt/dependencies/c-ares/;/opt/dependencies/c-ares/include;/opt/dependencies/protobuf" \
```

Comment:
In `/opt/dependencies/c-ares/;/opt/dependencies/c-ares/include`, only `/opt/dependencies/c-ares` should be enough because includes are searched in `include` subdirectory.

Reply:
But make sure we don't need any libraries prior removing `/opt/dependencies/c-ares/`.

Reply:
@bakhtin Can you explain more please? I suggest to replace `"/opt/dependencies/c-ares/;/opt/dependencies/c-ares/include;/opt/dependencies/protobuf"` with `"/opt/dependencies/c-ares;/opt/dependencies/protobuf"` because CMake will search in `include` subdirectory, not to remove `c-ares`.

Reply:
Ooops, sorry. I thought you suggested to leave just `/opt/dependencies/c-ares/include` and not `/opt/dependencies/c-ares`. It should be fine then.